### DISCUSSION TITLE
Off to a bad start!  Can't find UnitSquareMesh?

### DISCUSSION THREAD

**Discussion by bradlipovsky** [firedrakeproject/firedrake#2949] [Firedrake support]
First time using firedrake.  I can't seem to run the simplest demo I could find, the helmholtz solver. [The first line of code](https://github.com/firedrakeproject/firedrake/blob/master/demos/helmholtz/helmholtz.py.rst) is,

```
from firedrake import *
mesh = UnitSquareMesh(10, 10)
```
from which I am told,
```
NameError: name 'UnitSquareMesh' is not defined
```
I am running the latest docker image (created 4 days ago).

Brad

**Comment by connorjward** [comment]
Have you activated the virtual environment? You need to run `source <firedrake dir>/bin/activate`. I think `<firedrake dir>` is simply `/home/firedrake/firedrake`.

**Comment by bradlipovsky** [comment]
In my docker container, I ran
```
source /home/firedrake/firedrake/bin/activate
```
then I ran
```
python3
```
and
```
from firedrake import *
```
The result was this:
```
>>> from firedrake import *
[0]PETSC ERROR: ------------------------------------------------------------------------
[0]PETSC ERROR: Caught signal number 11 SEGV: Segmentation Violation, probably memory access out of range
[0]PETSC ERROR: Try option -start_in_debugger or -on_error_attach_debugger
[0]PETSC ERROR: or see https://petsc.org/release/faq/#valgrind and https://petsc.org/release/faq/
[0]PETSC ERROR: configure using --with-debugging=yes, recompile, link, and run 
[0]PETSC ERROR: to get more information on the crash.
[0]PETSC ERROR: Run with -malloc_debug to check if memory corruption is causing the crash.
application called MPI_Abort(MPI_COMM_WORLD, 59) - process 0
[unset]: PMIU_write error; fd=-1 buf=:cmd=abort exitcode=59 message=application called MPI_Abort(MPI_COMM_WORLD, 59) - process 0
:
system msg for write_line failure : Bad file descriptor
(firedrake) firedrake@81d88faf4a0d:~$ 
```

**Reply by connorjward** [reply]
This is curious. I have just tested the latest Docker container push (14 hours ago) and that worked fine. Could you try with an updated container?

**Reply by bradlipovsky** [reply]
I find the same result after running,
```
docker pull firedrakeproject/firedrake:latest
```

```
latest: Pulling from firedrakeproject/firedrake
Digest: sha256:6a2ee58b739b578c44c13196a45024fcf84938238c5045813409a972e46b88c0
Status: Image is up to date for firedrakeproject/firedrake:latest
docker.io/firedrakeproject/firedrake:latest
```

**Reply by connorjward** [reply]
I guess that your Docker is subtly different from mine somehow. What do you get if you run the following:
```
$ sudo apt update
$ sudo apt install gdb
$ source /home/firedrake/firedrake/bin/activate
$ gdb run --args python3 myfailingcode.py
```
(and enter `run` inside the `gdb` prompt).

Hopefully that will give more information about where the error is occurring.

You could also try installing Firedrake an alternative way. If you have Windows then [WSL](https://github.com/firedrakeproject/firedrake/wiki/Installing-on-Windows-Subsystem-for-Linux) works pretty well.
